drivers/firmware/scmi/shmem_internal.h

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+struct scmi_shmem_layout {
```

Comment:
this whole header file can probably be moved to `./include/zephyr/drivers/firmware/scmi/shmem.h`. This way, we can avoid introducing another header.